src/planscape/plan/views.py

```diff
@@ -240,34 +241,39 @@ def create_project(request: HttpRequest) -> HttpResponse:
                 (owner is not None and plan.owner is not None and owner.pk == plan.owner.pk)):
             raise ValueError(
                 "Cannot create project; plan is not owned by user")
-        
+
         project = Project.objects.create(owner=owner, plan=plan)
         _save_project_parameters(body, project)
         project.save()
         return HttpResponse(str(project.pk))
     except Exception as e:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
-    
+
+
 @csrf_exempt
 def update_project(request: HttpRequest) -> HttpResponse:
     try:
-        # Check that the user is logged in.
-        owner = _get_user(request)
+        if request.method == "PUT":
+            # Check that the user is logged in.
+            owner = _get_user(request)
 
-        body = json.loads(request.body)
-        project_id = body.get('id', None)
-        if project_id is None or not (isinstance(project_id, int)):
-            raise ValueError("Must specify project_id as an integer")
+            body = json.loads(request.body)
+            project_id = body.get('id', None)
+            if project_id is None or not (isinstance(project_id, int)):
+                raise ValueError("Must specify project_id as an integer")
 
-        project = Project.objects.get(id=project_id)
-        if project.owner != owner:
-            raise ValueError(
-                "You do not have permission to view this project.")
-        
-        project.priorities.clear()
-        _save_project_parameters(body, project)
-        project.save()
-        return HttpResponse(str(project.pk))
+            project = Project.objects.get(id=project_id)
+            if project.owner != owner:
+                raise ValueError(
+                    "You do not have permission to view this project.")
+
+            project.priorities.clear()
+            _save_project_parameters(body, project)
+            project.save()
+            return HttpResponse(str(project.pk))
+        else:
```

Comment:
Nit: the "else" is unnecessary as long as you unindent the next line.

Consider doing the error check first:

  if request.method != "PUT":
      raise KeyError(
                "HTTP methods other than PUT are not yet implemented")

Then you won't need as much indentation.
  if

Reply:
done!